go-controller/pkg/ovn/policy.go

```diff
@@ -1581,9 +1635,12 @@ func (oc *Controller) handlePeerPodSelectorAddUpdate(gp *gressPolicy, objs ...in
 		pods = append(pods, pod)
 	}
 	// If no IP is found, the pod handler may not have added it by the time the network policy handler
-	// processed this pod event. It will grab it during the pod update event to add the annotation,
-	// so don't log an error here.
-	if err := gp.addPeerPods(pods...); err != nil && !errors.Is(err, util.ErrNoPodIPFound) {
+	// processed this pod event. Ignore this error for Add events, but return it for update events to retry later.
+	// gressPolicy.addPeerPods must be called with networkPolicy RLock.
```

Comment:
This is a good place to discuss it
Now we have update local and peer pod handlers both for add and update pod events, I guess that was done before we had retry logic to make sure we retry on pod update if we fail to get pod information (pod ip or portUUID) on pod add event.
But now we will retry podAdd event for networkPolicy in any case, so I guess there is no need to have podUpdate handlers at all, wdyt?

Reply:
I think this makes sense. Especially since in the retry we get the latest version of the pod in the add. So we wouldn't be missing anything that would potentially be in the update.

Does gressPolicy.addPeerPods still need to be called with np RLock? I'm not seeing why.

Reply:
The main point to rlock network policy is to check if it was deleted. If it was, and you call `gressPolicy.addPeerPods` it will return error and you will retry infinitely

Reply:
in the end, I left add and update handlers, for local pods that is to make sure we update local pods asap after the pod was created, for peer pods that is required to make sure we update ips properly, I added some comments around that.
I also ignore errors on syncFunctions and allow individual objects to be retried later

Reply:
If we guaranteed handler order, so that addLogicalPort happened first and then peerPods, we would make sure we hit it on pod add and then maybe wouldn't need update anymore? I guess it makes sense to defer this for now.